ecs-init/volumes/types/types.go

```diff
@@ -19,23 +19,31 @@ type Volume struct {
 	Path      string
 	Options   map[string]string
 	CreatedAt string
-	Mounts    map[string]*string
+	Mounts    map[string]int
 }
 
 // Adds a new mount to the volume.
 // This method is not thread-safe, caller is responsible for holding any locks on the volume.
 func (v *Volume) AddMount(mountID string) {
 	if v.Mounts == nil {
-		v.Mounts = map[string]*string{}
+		v.Mounts = map[string]int{}
 	}
-	v.Mounts[mountID] = nil
+	v.Mounts[mountID] += 1
 }
 
 // Removes a mount from the volume.
 // This method is not thread-safe, caller is responsible for holding any locks on the volume.
 // Returns a bool indicating whether the mountID was found in mounts or not.
 func (v *Volume) RemoveMount(mountID string) bool {
 	_, exists := v.Mounts[mountID]
-	delete(v.Mounts, mountID)
-	return exists
+	if !exists {
+		return false
+	}
+
+	v.Mounts[mountID] -= 1
+	if v.Mounts[mountID] <= 0 {
```

Comment:
Just curious for my understanding, in what cases would there be a negative count of mounts for an EFS volume?

Reply:
That should never happen. Such a case would indicate a bug.

Reply:
Gotcha, thanks for the clarification. In that case, should this then be `v.Mounts[mountID] == 0`?

Reply:
I just wanted to be safe... I will update it to `== 0` so that any bugs don't get masked.

Reply:
Ah I see, if it's for safety then I'd prefer what you currently have

Reply:
Updated.

Reply:
Sync'd offline, we are both aligned with keeping `<= 0` here. So sorry for the back and forth and thanks a bunch for making this change!

Reply:
Reverted back to `<= 0` since `Mounts` field is exported.